src/Optimizer/Transformer/AutoExtensions.php

```diff
@@ -0,0 +1,609 @@
+     */
+    private function renderExtensionScripts(Document $document, array $extensionScripts)
+    {
+        $referenceNode = $this->getExtensionScriptsReferenceNode($document);
+
+        foreach ($extensionScripts as $extensionScript) {
+            if ($referenceNode && $referenceNode->nextSibling) {
+                $referenceNode->parentNode->insertBefore(
+                    $extensionScript,
+                    $referenceNode->nextSibling
+                );
+            } else {
+                $document->head->appendChild($extensionScript);
+            }
+
+            $referenceNode = $extensionScript;
+        }
+    }
+
+    /**
+     * Get the reference node to attach extension scripts to.
+     *
+     * @param Document $document Document to look for the reference node in.
+     * @return Element|null Reference node to use, or null if not found.
+     */
+    private function getExtensionScriptsReferenceNode(Document $document)
+    {
+        $referenceNode = $document->viewport ?: $document->charset;
+
+        if (! $referenceNode instanceof Element) {
+            $referenceNode = $document->head->firstChild;
+        }
+
+        if (! $referenceNode instanceof Element) {
+            return null;
+        }
+
+        // Try to detect the boilerplate style so we can append the scripts after that.
+        $remainingNode = $referenceNode->nextSibling;
+        while ($remainingNode) {
+            if (! $remainingNode instanceof Element) {
+                $remainingNode = $remainingNode->nextSibling;
+                continue;
+            }
+
+            if (
+                $remainingNode->tagName === Tag::STYLE
+                && $remainingNode->hasAttribute(Attribute::AMP_BOILERPLATE)
+            ) {
+                $referenceNode = $remainingNode;
+            } elseif (
+                $remainingNode->tagName === Tag::NOSCRIPT
+                && $remainingNode->firstChild instanceof Element
+                && $remainingNode->firstChild->tagName === Tag::STYLE
+                && $remainingNode->firstChild->hasAttribute(Attribute::AMP_BOILERPLATE)
+            ) {
+                $referenceNode = $remainingNode;
+            }
+
+            $remainingNode = $remainingNode->nextSibling;
+        }
+
+        return $referenceNode;
+    }
+
+    /**
+     * Maybe add a required extension to the list of extension scripts.
+     *
+     * @param Document $document          Document to render the extension scripts into.
+     * @param array    $extensionScripts  Existing list of extension scripts.
+     * @param string   $requiredExtension Required extension to check for.
+     * @return array Adapted list of extension scripts.
+     */
+    private function maybeAddExtension(Document $document, $extensionScripts, $requiredExtension)
+    {
+        if (in_array($requiredExtension, $this->configuration->get(AutoExtensionsConfiguration::IGNORED_EXTENSIONS))) {
```

Comment:
Array functions should always be used in their strict mode:
```suggestion
        if (in_array($requiredExtension, $this->configuration->get(AutoExtensionsConfiguration::IGNORED_EXTENSIONS), true)) {
```